include/swift/Remote/MetadataReader.h

```diff
@@ -524,6 +525,34 @@ class MetadataReader {
       return None;
 
     return start;
+#else
+    // All swift class instances start with an isa pointer,
+    // followed by the retain counts (which are the size of a long long)
+    size_t isaAndRetainCountSize = sizeof(StoredSize) + sizeof(long long);
+    size_t start = isaAndRetainCountSize;
+
+    auto superAddress = RemoteAddress(MetadataAddress + sizeof(StoredSize));
+    size_t super;
+    // Read the super pointer
+    if (!Reader->readInteger(superAddress, &super))
+      return 0;
+
+    while (super) {
+      auto instanceSizeAddress = RemoteAddress(super + (sizeof(StoredSize) * 3));
+      size_t instanceSize = 0;
+      if (!Reader->readInteger(instanceSizeAddress, 4, &instanceSize))
+        return 0;
+      // If there's a super, subtract the size contribution of the isa
+      // and retain counts
+      start += instanceSize - isaAndRetainCountSize;
+
+      // Lookup the next super
+      if (!Reader->readInteger(RemoteAddress(super + sizeof(StoredSize)), &super))
+        return 0;
+    }
+
+    return start;
+#endif
```

Comment:
Could we handle this by using `TargetClassMetadata<Runtime>`, instead of loading word-by-word?

Reply:
Thanks! It looks way cleaner now